model/task_lifecycle.go

```diff
@@ -637,31 +653,32 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 				return err
 			}
 		}
-		updates.BuildNewStatus = b.Status
-		updates.BuildComplete = true
+	}
+
+	// These are deliberately out of the buildComplete block to ensure versions
+	// are iterated so version and patch notifications can be sent out
+	if err = MarkVersionCompleted(b.Version, finishTime, updates); err != nil {
```

Comment:
does this mean that versions get marked complete before their tasks have finished running?

Reply:
Nope, the MarkVersionComplete method retains the old changes to status, the problem is that notifications and the task model have two different definitions of complete, an this method also handles the logic to log the notifications event.